src/bin/repl/mod.rs

```diff
@@ -21,6 +22,15 @@ const PREFIX: char = ':';
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 const PROMPT: &str = ">> ";
 
+macro_rules! execute {
+    ($fun:ident, $ty:path, $action:expr) => {
+        $action(unsafe {
+            let execute: unsafe extern "C" fn() -> $ty = std::mem::transmute($fun);
+            execute()
+        });
+    };
+}
```

Comment:
This is used like
```rust
execute!(fun, bool, |b| println!("=> {}", b)),
```
I think it would be easier and simpler to do `println!("=> {}", execute!(fun, bool))`, which means you don't need `$action` or to create a new closure.